drivers/net/ethernet/intel/idpf/idpf_main.c

```diff
@@ -278,4 +248,35 @@ static struct pci_driver idpf_driver = {
 	.remove			= idpf_remove,
 	.shutdown		= idpf_shutdown,
 };
-module_pci_driver(idpf_driver);
+
+/**
+ * idpf_driver_register - Device registration routine
+ * @void: void
+ *
+ * Returns 0 on success, otherwise function returned on failure
+ */
+static int __init idpf_driver_register(void)
+{
+	int err;
+
+	err = idpf_eth_idc_driver_register();
```

Comment:
The sequence should be reversed in my opinion because it is first a pci driver. So do the pci driver registration and the aux driver registration second.

Reply:
I believe this is how the ice driver also does. Also I don't see the issue from test either.